jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -103,13 +110,13 @@ public int size()
 
     public ListIterator<HttpField> listIterator()
     {
-        return new Itr();
+        return new ListItr();
     }
     
     
     public Stream<HttpField> stream()
     {
-        return StreamSupport.stream(Arrays.spliterator(_fields,0,_size),false);
+        return Arrays.stream(_fields).filter(f->f!=null);
```

Comment:
There can be nulls in the array beyond the _size index.  You can see the previous code used a spliterator to avoid the nulls....  I thought this might be simpler... but I'm not sure now....

Reply:
Ah I can use limit!